### PR TITLE
feat/create an endpoint to edit a location

### PR DESCRIPTION
**Pull request by Alao-Abiodun**
create an endpoint logic to edit a red-flag location record

### CODE REVIEW

File: src/services/incident.services.js [hunk 1]
```diff
@@ -30,6 +30,14 @@ const incidentServices = {
...
         incidents.push(incident);
         return incident;
+    },
+
+    // @describe update a red-flag 
+    // @public api/v1/red-flag/:id
+    updateRedFlagLocation (id, location) {
+        const redFlag = incidents.find(incident => incident.id === +id);
+        redFlag.location = location
+        return redFlag || {};
```
Comment (kenzdozz) [inline src/services/incident.services.js:43]: Why do you have `||` here?

Comment (Alao-Abiodun) [reply src/services/incident.services.js:43]: I am using the || in case nothing is found, then it should return an empty object.

Comment (kenzdozz) [reply src/services/incident.services.js:43]: Error should be thrown when someone attempts to update a record that doesn't exist.

The API call should return a 404 response

Comment (Alao-Abiodun) [reply src/services/incident.services.js:43]: > Error should be thrown when someone attempts to update a record that doesn't exist.
> 
> The API call should return a 404 response

Yes, I have added it in the controller if a particular is not found.

Comment (Alao-Abiodun) [reply src/services/incident.services.js:43]: services folder just work business logic, but the controller deal with result produce by the services logic.

File: src/controllers/incident.controller.js [hunk 2]
```diff
@@ -26,6 +26,13 @@ const incidentController = {
             }
             const data = incidentServices.addRegFlag(newflag);
             return responseHandler(res, 201, 'A new red-flag is created successfully...', data);
+    },
+
+    updateFlagLocation (req, res, next) {
```
Comment (kenzdozz) [inline src/controllers/incident.controller.js:31]: Where is the check if the item exists?

File: src/services/incident.services.js (hunk 1 above)
Comment (kenzdozz) [reply src/services/incident.services.js:43]: I get that, but the check if item exists is not in the controller

Comment (Alao-Abiodun) [reply src/services/incident.services.js:43]: > I get that, but the check if item exists is not in the controller

Is on the controller 

if (!id) {
            return next(new customError(404, 'Id is not found!'));
       }

**Review by kenzdozz — APPROVED** [review]

**Pull request merged**